src/App.js

```diff
@@ -10,10 +9,39 @@ const tabs = [
 ];
 /* eslint-enable */
 
-const App = () => (
-  <div className="App">
-    <h1>Selected tab is&nbsp;Tab 1</h1>
-  </div>
-);
+const preparedTabs = tabs.map((tab, index) => ({
+  ...tab,
+  id: index,
+}));
+
+class App extends React.Component {
+  state = {
+    currentIndex: 0,
+  };
+
+  onTabSelected = (index) => {
+    this.setState({
+      currentIndex: index,
+    });
+  }
```

Comment:
Додай перевірку, щоб не міняти стейт, якщо currentIndex не було змінено
```suggestion
  onTabSelected = (index) => {
    if (index === this.state.currentIndex) {
      return;
    }
  
    this.setState({
      currentIndex: index,
    });
  }
```

Reply:
Хороше зауваження від Артура